### DISCUSSION TITLE
ReactPy VS PyneCone

### DISCUSSION THREAD

**Discussion by saurabhwadekar** [idom-team/idom#988] [General]
ReactPy VS PyneCone

**Comment by rmorshea** [comment]
In general, my argument against PyneCone is that it relies on what (from what I can tell) is an outdated Model-View-Controller (MVC) strategy for state management. Over time, frontend developers have found that MVC just doesn't scale well. Wherever possible, ReactPy tries to replicate the well-liked and battle-hardened interfaces pioneered by ReactJS.

With that said, it seems like PyneCone, as it says, comes with a few more "batteries included" - for example, a database and built-in components based on Chakra UI. This comes at the cost of some flexibility though of course.

**Reply by rmorshea** [reply]
For example, due to ReactPy's modular construction, it can run in a variety of different contexts (e.g. [Django](https://github.com/reactive-python/reactpy-django) and [Jupyter Notebooks](https://github.com/reactive-python/reactpy-jupyter)). This allows you to write your components once, and use them anywhere.
